crates/e2e-test-utils/src/testsuite/examples.rs

```diff
@@ -88,24 +75,13 @@ async fn test_testsuite_complex_scenario() -> Result<()> {
 
     let test = TestBuilder::new(())
         .with_setup(setup)
-        .with_action(SubmitTransaction {
-            node_idx: 0,
-            raw_tx: vec![],
-            result_id: "tx1".to_string(),
-        })
-        .with_action(AdvanceBlock {
-            node_idx: 0,
-            transactions: vec![],
-            result_id: "block1".to_string(),
-        })
+        .with_action(SubmitTransaction { node_idx: 0, raw_tx: vec![] })
```

Comment:
it'd be great if we have action impl for `Fn(Env) -> Future` the we can do `move | env| async move {
  ...
}`

Reply:
nice, added here https://github.com/paradigmxyz/reth/pull/14773/commits/6bfa279d1719457c27435381644a862b20c3aa41